config/v1/types_cluster_version.go

```diff
@@ -290,6 +305,93 @@ type Release struct {
 // availableUpdates field is accurate and recent.
 const RetrievedUpdates ClusterStatusConditionType = "RetrievedUpdates"
 
+// ConditionalUpdate represents an update which is recommended to some
+// clusters on the version the current cluster is reconciling, but which
+// may not be recommended for the current cluster.
+// +k8s:deepcopy-gen=true
```

Comment:
I don't think this is necessary.

Reply:
I'm not entirely clear on what this is about, but my fuzzy impression was that we'd want it on structs that contained slices or other mutable properties.

Reply:
[here][1] is `ConditionalUpdate.DeepCopyInto` with handling for the `Risks` and `Conditions` slices, so that matches my fuzzy guess above^.  Convincing enough to leave this annotation in place?  Or do you think the deepcopy generator is smart enough to notice the slices and create the deepcopy code without needing the annotation hint?

[1]: https://github.com/openshift/api/commit/bc1d2cdfbf6ea0bc34e45429aeb196332fffe5c4#diff-ad96d6336ccdfffb67dcf48a7d0251b23c3960b5ab0d9db307eef19ec822d6e9R1161-R1180

Reply:
Checking to see what happens without this line:

```console
$ git diff
diff --git a/config/v1/types_cluster_version.go b/config/v1/types_cluster_version.gote str
index 62f79582..b64ca5cd 100644
--- a/config/v1/types_cluster_version.go
+++ b/config/v1/types_cluster_version.go
@@ -312,7 +312,6 @@ const RetrievedUpdates ClusterStatusConditionType = "RetrievedUpdates"
 // ConditionalUpdate represents an update which is recommended to some
 // clusters on the version the current cluster is reconciling, but which
 // may not be recommended for the current cluster.
-// +k8s:deepcopy-gen=true
 type ConditionalUpdate struct {
        // release is the target of the update.
        // +required
$ hack/update-deepcopy.sh 
go: downloading k8s.io/gengo v0.0.0-20201214224949-b6c5ce23f027
...
Generating deepcopy funcs
$ git status
On branch targeted-edge-blocking
Your branch is up to date with 'origin/targeted-edge-blocking'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   config/v1/types_cluster_version.go

no changes added to commit (use "git add" and/or "git commit -a")
```

So looks plausible for "deepcopy generator is smart enough to find this without the hint".  I've pushed bc1d2cdf -> 37280ee5 to drop this line.